syncd/GlobalSwitchId.cpp

```diff
@@ -33,4 +37,10 @@ void GlobalSwitchId::setSwitchId(
             sai_serialize_object_id(gSwitchId).c_str());
 #endif
 
+#ifdef MDIO_ACCESS_USE_NPU
+    if (mdioSwitchId == SAI_NULL_OBJECT_ID)
+    {
+        mdioSwitchId = switchRid;
+    }
+#endif
```

Comment:
this mdioSwitchId  needs to be passed to this class that you will create in the constructor, then remove global definition